import { AxiosError } from "axios";
import api from "../../core/api";
import { LoginRequest, LoginResponse } from "./login.interface";

export const login = async (
  credentials: LoginRequest
): Promise<LoginResponse> => {
  try {
    const response = await api.post<LoginResponse>("/auth/login", credentials);
    console.log(response.data)
    return response.data;
  } catch (error) {
    if (error instanceof AxiosError && error.response) {
      throw new Error(error.response.data.message || "Erro ao realizar login");
    }
    throw new Error("Erro de conex√£o com o servidor");
  }
};